cala-ledger/src/ledger/mod.rs

```diff
@@ -222,6 +222,82 @@ impl CalaLedger {
         Ok(transaction)
     }
 
+    pub async fn void_transaction(
+        &self,
+        original_tx_id: TransactionId,
+    ) -> Result<Transaction, LedgerError> {
+        let mut db = LedgerOperation::init(&self.pool, &self.outbox).await?;
+        let transaction = self.void_transaction_in_op(&mut db, original_tx_id).await?;
+        db.commit().await?;
+        Ok(transaction)
+    }
+
+    #[instrument(
+        name = "cala_ledger.transaction_void",
+        skip(self, db)
+        fields(transaction_id, external_id)
+        err
+    )]
+    pub async fn void_transaction_in_op(
+        &self,
+        db: &mut LedgerOperation<'_>,
+        original_tx_id: TransactionId,
+    ) -> Result<Transaction, LedgerError> {
+        let new_tx_id = TransactionId::new();
+
+        let new_entries = self
+            .entries()
+            .new_entries_for_voided_tx(original_tx_id, new_tx_id)
+            .await?;
+
+        let transaction = self
+            .transactions()
+            .create_voided_tx_in_op(
+                db,
+                original_tx_id,
+                new_tx_id,
+                new_entries.iter().map(|entry| entry.id),
+            )
+            .await?;
+
+        let span = tracing::Span::current();
+        span.record("transaction_id", transaction.id().to_string());
+        span.record("external_id", &transaction.values().external_id);
+
+        let entries = self.entries.create_all_in_op(db, new_entries).await?;
```

Comment:
these is something inconsistent here:
```
  let new_entries = self
            .entries()
```
there you are using `entries()` and here `entries`. Also why do we need to go to `entries` twice? Can't we create + persist in 1 round trip to that service?